lib/sudo-exec/src/event.rs

```diff
@@ -32,6 +51,27 @@ impl<T> EventHandler<T> {
         self.callbacks.push(callback);
     }
 
+    /// Set the handler for `SIGNAL` to be polled for read events and set `callback` to be called
+    /// if the handler is ready.  
+    pub(crate) fn set_signal_callback<const SIGNAL: c_int>(&mut self, callback: Callback<T>) {
+        let id = EventId(self.callbacks.len());
+        let Some(handler) = self.signal_handlers.get(&SIGNAL) else { 
+            return;
+        };
+        self.poll_set.add_fd_read(id, handler);
+        self.callbacks.push(callback);
+    }
+
+    /// Receive the information related to the arrival of a signal with number `SIGNAL`. Return
+    /// `None` if the signal is not in [`SIGNALS`].
+    ///
+    /// Calling this function will block until a signal with number `SIGNAL` arrives.
+    pub(crate) fn recv_signal_info<const SIGNAL: c_int>(
```

Comment:
same here. using a `SIGNAL` that's not in the `SIGNALS` set feels like a programmer error.

maybe at this point it would make sense to create an enumeration of the signals we want to handle. `signal_handlers` would always contain all the variants of that enum so there should be no runtime error = no possibility of misuse.